experimental/packages/opentelemetry-sdk-metrics-base/src/export/MetricReader.ts

```diff
@@ -91,15 +92,15 @@ export abstract class MetricReader {
   /**
    * Collect all metrics from the associated {@link MetricProducer}
    */
-  async collect(options?: ReaderCollectionOptions): Promise<MetricData[]> {
+  async collect(options?: ReaderCollectionOptions): Promise<Maybe<ResourceMetrics>> {
```

Comment:
This change seems unrelated to the rest of the PR. What motivated it?

Reply:
Think it's a remnant when I used the top level `MetricsData` object containing an array of resources initially.
On shutdown MetricReader returned an empty array before and in the successful case all the MetricData points contained the same resource.

However now it is only collecting from the producer which returns a single `ResourceMetrics`. So either it returns an array of `ResourceMetrics` here or undefined. As it stands currently the exporters are called with a single `ResourceMetrics` :thinking: Perhaps we ought to export `ResourceMetrics[]` instead to keep in line with [protos](https://github.com/open-telemetry/opentelemetry-proto/blob/main/opentelemetry/proto/metrics/v1/metrics.proto#L43)? But it'll be a singular array every.

Reply:
I missed that it is not an array anymore. I think this is fine.